core/src/main/scala/org/apache/spark/metrics/MetricsSystem.scala

```diff
@@ -103,6 +103,14 @@ private[spark] class MetricsSystem private (
     sinks.foreach(_.start)
   }
 
+  // Same as start but this method only registers sinks
```

Comment:
Would you please explain why only registering sinks could solve the problem here?

Reply:
What I am trying to do is to avoid the registration of the "static metrics", for CodeGeneration and HiveExternalCatalog and also for JVM.